scripts/spark/added/launch.sh

```diff
@@ -9,6 +9,22 @@ function check_reverse_proxy {
     fi
 }
 
+function handle_term {
+    echo Received a termination signal
+    # If we've saved a PID for a subprocess, kill that first before
+    # trying to delete the cluster
+    if [ -n "$PID" ]; then
+        echo "Stopping subprocess $PID"
+        kill -TERM $PID
+        wait $PID
+        echo "Subprocess stopped"
+    fi
+    exit 0
+}
+
+
+trap handle_term TERM INT
```

Comment:
Hard to find docs on standard signals handled by Spark. I could experiment with spark running standalone and see what else it responds to ... maybe SIGHUP, there is a log handler for that.

If our handler reflects the signals handled by Spark, that might be good.  But this is a definite improvement over what we've got now.